geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -354,6 +350,9 @@ class SelectionMapFragment(
             itemsByFeatureId[featureId] = item
             points.add(point)
         }
+        selectedFeatureViewModel.getSelectedFeatureId()?.let {
```

Comment:
It looks like we're missing test coverage for this change (and the original placement of the code). I guess the code itself is for two things:

1. Keeping the currently selected item selected if the items change (not really a use case we currently have)
2. Keeping the currently selected item selected when the Fragment is recreated

I've added a test for **2** and verified that it forces us to move the code around by moving it back and seeing it fail.

In writing that test, I then realized there was a problem where if we started the fragment with a selection, selecting a new item and then selecting again would always result in the initial selection being displayed. I wrote a test for that case and then realized we don't need this code at all. That's all committed here.